features/step_definitions/generic_api.rb

```diff
@@ -25,3 +25,11 @@
 And(/^a message for failure is given "([^"]*)"$/) do |failure_reason|
   assert_match(failure_reason, @response)
 end
+
+And(/^I know the borrower id$/) do
+  deed_end_point = @response.body.to_s
+  @response = HTTP.get(deed_end_point, json: @deed_hash)
```

Comment:
Just a thought that I'm not sure if you need to include the 'json: @deed_hash' bit in the get request as I don't think you need to pass anthing through in the payload with a get request. Might want to just double check with a dev though.